include/swift/Option/Options.td

```diff
@@ -313,6 +313,9 @@ def no_color_diagnostics : Flag<["-"], "no-color-diagnostics">,
 def debug_diagnostic_names : Flag<["-"], "debug-diagnostic-names">,
   Flags<[FrontendOption, DoesNotAffectIncrementalBuild, HelpHidden]>,
   HelpText<"Include diagnostic names when printing">;
+def enable_descriptive_diagnostics : Flag<["-"], "enable-experimental-descriptive-diagnostics">,
```

Comment:
It's a little funny to have this have a different internal name than the public name, but I guess since the missing word is "experimental" it's okay.